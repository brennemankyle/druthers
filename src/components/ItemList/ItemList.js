import React from 'react'
import PropTypes from 'prop-types'
import AppPropTypes from '../../utils/AppPropTypes'
import withKeys from '../../utils/withKeys'

const noop = () => {}

let ItemList = (props) => {
  let styles = withKeys(props, 'styles_')
  let Item = props.Item

  let renderItem = (item) =>
    <Item
      item={item}
      removable={props.removable}
      key={item.value}
      svg_Remove={props.svg_Remove}
      {...styles} />

  let hasItems = !!props.itemList.length
  let onMouseDown = hasItems ? props.onClick : noop
  let onMouseOver = hasItems ? props.onMouseOver : noop
  let onMouseOut = hasItems ? props.onMouseOut : noop

  return (
    <ul className={props.className} onMouseDown={onMouseDown} onPointerDown={onMouseDown} onTouchStart={onMouseDown} onMouseOver={onMouseOver} onMouseOut={onMouseOut}>
      {props.itemList.map(renderItem)}
      {!hasItems && props.noItemsText && <Item
        item={{value: '', label: props.noItemsText}}
        removable={false}
        {...styles} />}
    </ul>
  )
}

ItemList.defaultProps = {
  removable: false,
}

ItemList.propTypes = {
  itemList: AppPropTypes.itemList.isRequired,
  removable: PropTypes.bool,
  onClick: PropTypes.func,
  noItemsText: PropTypes.string,
  onMouseOver: PropTypes.func,
  onMouseOut: PropTypes.func,
  Item: AppPropTypes.element.isRequired,
  svg_Remove: AppPropTypes.element,
  ...AppPropTypes.styles,
}

export default ItemList
