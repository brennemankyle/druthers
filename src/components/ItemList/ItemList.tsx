import React, { MouseEventHandler, ReactElement } from "react";
import { Item } from "../../utils/SelectTypes";
import { withKeys, noop } from "../../utils/utils";

const preventBlur = (e) => e.preventDefault();

interface Props {
  className: string;
  itemList: Item[];
  removable: boolean;
  onClick: MouseEventHandler<HTMLUListElement>;
  noItemsText: string;
  onMouseOver: MouseEventHandler<HTMLUListElement>;
  onMouseOut: MouseEventHandler<HTMLUListElement>;
  Item: ReactElement;
  svg_Remove: ReactElement;
}

function ItemList(props: Props) {
  let styles = withKeys(props, "styles_");
  let Item = props.Item;

  let renderItem = (item) => (
    <Item
      item={item}
      removable={props.removable}
      key={item.value || item.label}
      svg_Remove={props.svg_Remove}
      {...styles}
    />
  );

  let hasItems = !!props.itemList.length;
  let onClick = hasItems ? props.onClick : noop;
  let onMouseOver = hasItems ? props.onMouseOver : noop;
  let onMouseOut = hasItems ? props.onMouseOut : noop;

  return (
    <ul
      className={props.className}
      onClick={onClick}
      onMouseDown={preventBlur}
      onTouchStart={preventBlur}
      onMouseOver={onMouseOver}
      onMouseOut={onMouseOut}
    >
      {props.itemList.map(renderItem)}
      {!hasItems && props.noItemsText && (
        <Item
          item={{ value: "", label: props.noItemsText }}
          removable={false}
          {...styles}
        />
      )}
    </ul>
  );
}

ItemList.defaultProps = {
  removable: false,
};

export default ItemList;
