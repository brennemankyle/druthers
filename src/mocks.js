let mockStyles = {
  colors: {}
}

let mockItem = {
  value: '42',
  label: 'Meaning of life',
}

let mockItemList = [
  {value: '1', label: 'Item 1'},
  {value: '2', label: 'Item 2'},
  {value: '3', label: 'Item 3'},
  {value: '4', label: 'Item 4'},
  {value: '5', label: 'Item 5'},
]

export { mockStyles, mockItem, mockItemList }
