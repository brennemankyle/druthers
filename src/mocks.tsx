import React from "react";
import defaultProps from "./utils/defaultProps";
import { withKeys, noop } from "./utils/utils";

let mockStyles = {
  styles_width: 600,
  styles_multiple: false,
  styles_disabled: false,
  styles_hasSelection: false,
  styles_hasOptions: true,
  styles_optionHighlighted: [],
  styles_rightToLeft: false,
  ...withKeys(defaultProps, "styles_"),
};

let mockItem = {
  value: "42",
  label: "Meaning of life",
};

let mockItemList = [
  { value: "1", label: "Item 1" },
  { value: "2", label: "Item 2" },
  { value: "3", label: "Item 3" },
  { value: "4", label: "Item 4" },
  { value: "5", label: "Item 5" },
];

let mockEvent = {
  preventDefault: noop,
};

let MockElement = (props: any) => <div data-props={props} />;
let MockInput = (props: any) => <input type="text" data-props={props} />;
let MockSvg = () => (
  <svg>
    <path d="M0 0L5 5Z" />
  </svg>
);

let mockRect = {
  top: 10,
  bottom: 110,
  left: 20,
  right: 220,
  width: 200,
  height: 100,
  x: 15,
  y: 25,
};

export {
  mockStyles,
  mockItem,
  mockItemList,
  mockEvent,
  MockElement,
  MockInput,
  MockSvg,
  mockRect,
};
