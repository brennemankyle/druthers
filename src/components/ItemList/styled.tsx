import React from "react";
import styled from "@emotion/styled";
import ItemList from "./ItemList";

const SelectionList = styled(ItemList)(
  (props) => `
  ${
    props.styles_multiple
      ? `display: inline;`
      : `display: inline-block;
      &:not(:empty) {
        width: 100%;
      }`
  }
  list-style-type: none;
  padding: 0;
  margin: 0;
  color: ${props.styles_colors_text};
`
);

const OptionList = styled(ItemList)(
  (props) => `
  color: ${props.styles_colors_text};
  list-style-type: none;
  padding: 0;
  margin: 0;
  margin-block-start: 0;
  margin-block-end: 0;
  border-color: ${props.styles_colors_secondary};
  border-width: ${props.styles_borderWidth};
  border-style: ${props.styles_borderStyle};
  border-radius: ${props.styles_borderRadius};

  .truncate-show, .truncate-hide {
    cursor: pointer;
  }
`
);

export { SelectionList, OptionList };
