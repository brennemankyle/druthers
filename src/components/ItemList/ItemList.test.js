import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { mockStyles, mockItemList, MockElement } from '../../mocks'
import ItemList from './ItemList'

const onClick = jest.fn()
const onMouseOver = jest.fn()
const onMouseOut = jest.fn()

it('renders', () => {
  const wrapper = shallow(<ItemList
    {...mockStyles}
    itemList={mockItemList}
    Item={MockElement} />)

  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders removable', () => {
  const wrapper = shallow(<ItemList
    {...mockStyles}
    itemList={mockItemList}
    Item={MockElement}
    removable={true} />)

  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders no items', () => {
  const wrapper = shallow(<ItemList
    {...mockStyles}
    itemList={[]}
    Item={MockElement}
    removable={true}
    noItemsText="None"
    onClick={onClick}
    onMouseOver={onMouseOver} />)

  wrapper.find('ul').simulate('click')
  wrapper.find('ul').simulate('mouseOver')

  expect(onClick).not.toBeCalled()
  expect(onMouseOver).not.toBeCalled()
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('should click', () => {
  const wrapper = shallow(<ItemList
    {...mockStyles}
    itemList={mockItemList}
    Item={MockElement}
    onClick={onClick} />)

  wrapper.find('ul').simulate('click')

  expect(onClick).toBeCalled()
})

it('should mouseOver', () => {
  const wrapper = shallow(<ItemList
    {...mockStyles}
    itemList={mockItemList}
    Item={MockElement}
    onMouseOver={onMouseOver} />)

  wrapper.find('ul').simulate('mouseOver')

  expect(onMouseOver).toBeCalled()
})

it('should mouseOver', () => {
  const wrapper = shallow(<ItemList
    {...mockStyles}
    itemList={mockItemList}
    Item={MockElement}
    onMouseOut={onMouseOut} />)

  wrapper.find('ul').simulate('mouseOut')

  expect(onMouseOut).toBeCalled()
})
