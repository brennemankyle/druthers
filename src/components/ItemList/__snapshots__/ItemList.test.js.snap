// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<ul>
  <MockElement
    item={
      Object {
        "label": "Item 1",
        "value": "1",
      }
    }
    key="1"
    removable={false}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
  <MockElement
    item={
      Object {
        "label": "Item 2",
        "value": "2",
      }
    }
    key="2"
    removable={false}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
  <MockElement
    item={
      Object {
        "label": "Item 3",
        "value": "3",
      }
    }
    key="3"
    removable={false}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
  <MockElement
    item={
      Object {
        "label": "Item 4",
        "value": "4",
      }
    }
    key="4"
    removable={false}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
  <MockElement
    item={
      Object {
        "label": "Item 5",
        "value": "5",
      }
    }
    key="5"
    removable={false}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
</ul>
`;

exports[`renders no items 1`] = `
<ul
  onMouseDown={[Function]}
  onMouseOver={[Function]}
>
  <MockElement
    item={
      Object {
        "label": "None",
        "value": "",
      }
    }
    removable={false}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
</ul>
`;

exports[`renders removable 1`] = `
<ul>
  <MockElement
    item={
      Object {
        "label": "Item 1",
        "value": "1",
      }
    }
    key="1"
    removable={true}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
  <MockElement
    item={
      Object {
        "label": "Item 2",
        "value": "2",
      }
    }
    key="2"
    removable={true}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
  <MockElement
    item={
      Object {
        "label": "Item 3",
        "value": "3",
      }
    }
    key="3"
    removable={true}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
  <MockElement
    item={
      Object {
        "label": "Item 4",
        "value": "4",
      }
    }
    key="4"
    removable={true}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
  <MockElement
    item={
      Object {
        "label": "Item 5",
        "value": "5",
      }
    }
    key="5"
    removable={true}
    styles={
      Object {
        "borderRadius": ".2em",
        "checkRadio": Object {
          "borderWidth": "2px",
          "labelMargin": ".2em",
          "marginBetween": "1em",
          "paddingBottom": ".2em",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": ".2em",
        },
        "colors": Object {
          "background": "white",
          "disabled": "#ECECEC",
          "highlight": "lightblue",
          "primary": "black",
          "secondary": "#C3C3C3",
          "warning": "#CD5C5C",
        },
        "disabled": false,
        "fontSize": "1em",
        "hasOptions": true,
        "hasSelection": false,
        "icon": Object {
          "width": "1em",
        },
        "multiple": false,
        "option": Object {
          "paddingBottom": ".25em",
          "paddingLeft": ".25em",
          "paddingRight": ".25em",
          "paddingTop": ".25em",
        },
        "optionHighlighted": Array [],
        "paddingBottom": ".25em",
        "paddingLeft": ".25em",
        "paddingRight": ".25em",
        "paddingTop": ".25em",
        "rightToLeft": false,
        "search": Object {
          "width": "10em",
        },
        "selection": Object {
          "paddingBottom": "0",
          "paddingLeft": ".2em",
          "paddingRight": ".2em",
          "paddingTop": "0",
        },
        "width": "600px",
      }
    }
  />
</ul>
`;
